apps/rush-lib/src/scripts/install-run-common.ts

```diff
@@ -0,0 +1,299 @@
+
+/**
+ * Run "npm install" in the package install folder.
+ */
+function installPackage(packageInstallFolder: string, name: string, version: string): void {
+  try {
+    console.log(`Installing ${name}...`);
+    const npmPath: string = getNpmPath();
+    childProcess.execSync(`"${npmPath}" install`, { cwd: packageInstallFolder });
+    console.log(`Successfully installed ${name}@${version}`);
+  } catch (e) {
+    throw new Error(`Unable to install package: ${e}`);
+  }
+}
+
+/**
+ * Try to resolve the specified binary in an installed package.
+ */
+function findBinPath(packageInstallFolder: string, name: string, binName: string): string {
+  try {
+    const packagePath: string = path.resolve(packageInstallFolder, NODE_MODULES_FOLDER_NAME, name);
+    const packageJsonPath: string = path.resolve(packagePath, PACKAGE_JSON_FILENAME);
+    const packageJson: IPackageJson = require(packageJsonPath);
+    if (!packageJson.bin) {
+      throw new Error('No binaries are specified for package.');
+    } else {
+      const binValue: string = packageJson.bin[binName];
+      if (!binValue) {
+        throw new Error(`Binary ${binName} is not specified in the package's package.json`);
+      } else {
+        const resolvedBinPath: string = path.resolve(packagePath, binValue);
+        if (!fs.existsSync(resolvedBinPath)) {
+          throw new Error('The specified binary points to a path that does not exist');
+        } else {
+          return resolvedBinPath;
+        }
+      }
+    }
+  } catch (e) {
+    throw new Error(`Unable to find specified binary "${binName}": ${e}`);
+  }
+}
+
+/**
+ * Write a flag file to the package's install directory, signifying that the install was successful.
+ */
+function writeFlagFile(packageInstallFolder: string): void {
+  try {
+    const flagFilePath: string = path.join(packageInstallFolder, INSTALLED_FLAG_FILENAME);
+    fs.writeFileSync(flagFilePath, process.version);
+  } catch (e) {
+    // Ignore
+  }
+}
+
+export function installAndRun(
+  nodePath: string,
+  packageName: string,
+  packageVersion: string,
+  packageBinName: string,
+  packageBinArgs: string[]
+): void {
+  const rushJsonFolder: string = findRushJsonFolder();
+  const rushCommonFolder: string = path.join(rushJsonFolder, 'common');
+  const packageInstallFolder: string = ensureAndResolveFolder(
+    rushCommonFolder,
+    'temp',
+    'install-run',
+    `${packageName}@${packageVersion}`
+  );
+
+  if (!isPackageAlreadyInstalled(packageInstallFolder)) {
+    // The package isn't already installed
+    cleanInstallFolder(rushCommonFolder, packageInstallFolder);
+    copyNpmrcIfItExists(rushCommonFolder, packageInstallFolder);
+    createPackageJson(packageInstallFolder, packageName, packageVersion);
+    installPackage(packageInstallFolder, packageName, packageVersion);
+    writeFlagFile(packageInstallFolder);
+  }
+
+  const binPath: string = findBinPath(packageInstallFolder, packageName, packageBinName);
+  childProcess.spawnSync(
```

Comment:
>spawnSync [](start = 15, length = 9)

We should pass along the exit code from spawn.  Might be useful to compare with the [rushx implementation](https://github.com/Microsoft/web-build-tools/blob/master/apps/rush-lib/src/cli/RushXCommandLine.ts#L94)